chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/ChaosMonkeyRequestScope.java

```diff
@@ -109,8 +113,9 @@ private boolean isTrouble() {
         >= chaosMonkeySettings.getAssaultProperties().getLevel();
   }
 
-  private boolean isEnabled() {
-    return this.chaosMonkeySettings.getChaosMonkeyProperties().isEnabled();
+  private boolean isEnabled(MetricType type, String name) {
+    return this.chaosMonkeySettings.getChaosMonkeyProperties().isEnabled()
+        && chaosToggles.isEnabled(chaosToggles.nameMapper(type, name));
```

Comment:
This is the key change: when we check if Chaos Monkey is enabled, we also check that the corresponding toggle is enabled.

The default implementation is `true` to existing users won't need to change.

Note that we are now passing in the MetricType so we can include that in the `nameMapper` and use it to group the toggle. I don't love that we are reusing the MetricType enum, since we aren't using it for metrics purposes, but I like that it does mostly align.